ocaml/xapi/xapi_pgpu.ml

```diff
@@ -48,9 +48,17 @@ let maybe_fetch_compatibility_metadata ~__context ~pgpu_pci =
     []
 
 let populate_compatibility_metadata ~__context ~pgpu ~pgpu_pci =
-  let () = Db.PGPU.set_compatibility_metadata ~__context ~self:pgpu
-    ~value:(maybe_fetch_compatibility_metadata ~__context ~pgpu_pci)
```

Comment:
Can we now delete the `maybe_fetch_compatibility_metadata` function above?

Reply:
It is used in one more place. Will take a look.

Reply:
https://github.com/xapi-project/xen-api/blob/vgpu-migration/ocaml/xapi/xapi_pgpu.ml#L67

Reply:
`maybe_fetch_compatibility_metadata` is used in the `create` function. I also don't like the code duplication between `populate_compatibility_metadata` and `fetch_compatibility_metadata` but reducing this would lead to more code in other places. So I think I would keep it.